apps/src/componentLibrary/typography/SpecificComponentsGenerator.tsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import Typography from './index';
+import {
+  SemanticTag,
+  VisualApproach,
+  SpecificTypographyElementProps
+} from './types';
+
+interface SpecificTypographyElementToGenerateTemplate {
```

Comment:
minor: if it helps to slim this down, I feel like we can probably remove the `Specific` naming from these types/classes, e.g. `TypographyElementToGenerateTemplate`, `ComponentsGenerator`, even `TypographyElementProps`, etc